code/modules/hydroponics/grown.dm

```diff
@@ -72,15 +86,93 @@
 		return
 	return
 
+
+// Various gene procs
+/obj/item/weapon/reagent_containers/food/snacks/grown/attack_self(mob/user)
+	if(seed.get_gene(/datum/plant_gene/trait/squash))
+		squash(user)
+	..()
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/throw_impact(atom/hit_atom)
+	if(!..() && seed.get_gene(/datum/plant_gene/trait/squash)) //was it caught by a mob?
+		squash(hit_atom)
+
+/obj/item/weapon/reagent_containers/food/snacks/grown/proc/squash(atom/target)
+	var/turf/T = get_turf(target)
+	if(ispath(splat_type, /obj/effect/decal/cleanable/plant_smudge))
+		if(filling_color)
+			var/obj/O = new splat_type(T)
+			O.color = filling_color
+			O.name = "[name] smudge"
+	else if(splat_type)
+		new splat_type(T)
+
+	if(trash)
+		if(ispath(trash, /obj/item/weapon/grown) || ispath(trash, /obj/item/weapon/reagent_containers/food/snacks/grown))

```

Comment:
same usage as istype() except that it will silently fail with a bad typepath, returning 0 instead.